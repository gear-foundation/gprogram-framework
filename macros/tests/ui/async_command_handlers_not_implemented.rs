use gprogram_framework_macros::{command_handlers, gprogram};

#[gprogram]
mod gprogram {
    use super::*;

    command_handlers!(
        async fn do_this() -> Result<(), ()> {
            Ok(())
        }
    );
}

fn main() {}
