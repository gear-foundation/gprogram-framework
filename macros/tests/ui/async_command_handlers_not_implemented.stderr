error: Async command handlers are not implemented. Please use sync ones
  --> tests/ui/async_command_handlers_not_implemented.rs:8:9
   |
8  | /         async fn do_this() -> Result<(), ()> {
9  | |             Ok(())
10 | |         }
   | |_________^
